### DISCUSSION TITLE
How to overide db for testing CLI commands

### DISCUSSION THREAD

**Discussion by uday12kumar** [fastapi/typer#722] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer


app = typer.Typer()

def get_db():
    return sessionLocal()

@app.command()
def tasks():
    session = get_db()
    tasks_ = session.scalars(select(Tasks)).all()
    typer.echo(f"tasks count = {len(tasks_)}")

how to over-ride get_db method.
```

### Description

I have created commands which use database session to fetch the data from the db. Is there any way I can test the commands by over-riding the database session for testing the typer commands

### Operating System

Linux

### Typer Version

0.9.0

### Python Version

3.8

**Comment by kinuax** [comment]
Mocking the database is one way. There is a Real Python post [Understanding the Python Mock Object Library](https://realpython.com/python-mock-library/).
